nemo/export/trt_llm/nemo/convert.py

```diff
@@ -262,6 +262,9 @@ def split_and_save_weight(tp_rank, saved_dir, split_factor, key, vals, storage_t
         or "mlp.linear_fc1.weight" in key
         or "mlp.linear_fc1.bias" in key
     ):
+        if config["decoder_type"] == "chatglm":
```

Comment:
Rather than using a specific decoder type here, can we use/add a parameter such as fc_layer_type (please feel free to pick a better variable name based on it category)? I understand that chatglm is using a specific format to store the fc layer here but maybe in the future, another model might do the same and we can use it for that model too.